core/block_svg.js

```diff
@@ -619,9 +619,22 @@ Blockly.BlockSvg.prototype.tab = function(start, forward) {
   var target = list[forward ? i + 1 : i - 1];
   if (!target) {
     // Ran off of list.
-    var parent = this.getParent();
-    if (parent) {
-      parent.tab(this, forward);
+    // If there is an output, tab up to that block.
+    var outputBlock = this.outputConnection && this.outputConnection.targetBlock();
+    if (outputBlock) {
```

Comment:
I believe because you've basically "bottomed out", and the only way to go is back "up" to the output connection.